crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/FindCommand.java

```diff
@@ -121,13 +121,24 @@ public void write(OutputStream os) throws IOException {
                     writer.flush();
                 } finally {
                     streamResponse.documentStream.close();
+                    if (context != null) {  
+                        context.endRequestMonitoring();
+                    }
+                    // TODO: What if there is IOException?
                 }
             }
         };
     }
 
     @Override
     public CallStatus run() {
+        // Handle metrics only for streaming requests at this level, simple/bulk find request's 
+        // metrics will be handled at mediator level. This is because behavior is different
+        // if streaming or not. Exceptions always end request monitoring, but streaming is ended
+        // only when stream is fully written.
+        if(stream) {
+            context = metrics.startStreamingEntityRequest("find", entity, version);
+        }
```

Comment:
We don't need to pollute the HTTP layer with this here, we can still track it in the Mediator. We just have to listen for when the streaming results are depleted.